### ISSUE TITLE
How to solve this issue of The session or access key being used is invalid, please run "code-push-standalone login" again.

### ISSUE THREAD

**Issue by Gautham495** [microsoft/code-push-server#10]
<img width="946" alt="Screenshot 2024-10-05 at 8 13 48 PM" src="https://github.com/user-attachments/assets/5a1447d0-8010-4b95-82cf-0dd00bd69b11">

I have done all the things mentioned in the readme and i am stuck here. Please solve this.

**Comment by cleberhensel** [comment]
I had the same problem.
In my case, it was because I was running the application incorrectly.

I started running something like:

`yarn clean && yarn build && yarn start:env`

But you need to have filled in 1 set of keys
either GitHub Or Microsoft in `./api/.env`

GITHUB_CLIENT_ID
GITHUB_CLIENT_SECRET
MICROSOFT_CLIENT_ID
MICROSOFT_CLIENT_SECRET

**Comment by icarusdust** [comment]
I was having the same issue, stopped and restarted it, it helped but not completely. Now, I am getting the same error in the server logs. Here are the steps that I followed:

1. I have `.env` in `./api` with `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `MICROSOFT_CLIENT_ID` and `MICROSOFT_CLIENT_SECRET` set 
2. While server is up and running, I run `code-push-standalone login` which opens a browser to authenticate, when I am clicking on Github, or Microsoft, it redirects to authorize the action, then it throws `Internal Sever Error`. Upon checking the logs, it says `Error: connect ECONNREFUSED 127.0.0.1:10002`. I added `HTTP_PROXY` in .env to fix it, then the error changing to the same message mentioned in the original question.

**Comment by Jeyoung-Park** [comment]
@icarusdust 
i am using azurite by visual studio code.
port 10002 was used by azurite:table port by defaultm so i changed to 10003 and then it worked.

![image](https://github.com/user-attachments/assets/7534056d-ea3d-4f03-b643-49837a584a53)

**Comment by icarusdust** [comment]
@Jeyoung-Park thanks, I have already updated it to 10003 and synced it but still throwing the same error. Wondering if you configured anything specifically inside `.env`

**Comment by Jeyoung-Park** [comment]
> @Jeyoung-Park thanks, I have already updated it to 10003 and synced it but still throwing the same error. Wondering if you configured anything specifically inside `.env`

@icarusdust 

i just used GITHUB_CLIENT_ID and GITHUB_CLIENT_SECRET
I didn't use MICROSOFT_CLIENT_ID and MICROSOFT_CLIENT_SECRET

**Comment by imanshul** [comment]
I'm also facing the same issue.

Error on page at http://localhost:3000/auth/callback/github?code=someValue&state=someValue
> Internal Server Error

Error in logs:
> Error: connect ECONNREFUSED 127.0.0.1:10002

My .env has following values:

EMULATED=true
SERVER_URL=http://localhost:3000
GITHUB_CLIENT_ID=someValue
GITHUB_CLIENT_SECRET=someValue
LOGGING=true

**Comment by imanshul** [comment]
I'm able to resolve the **Error: connect ECONNREFUSED 127.0.0.1:10002** issue by installing azurite and running the server locally.

Just modify .env to have below config, which is default for Azurite (emulator for local)

AZURE_STORAGE_ACCOUNT=devstoreaccount1
AZURE_STORAGE_ACCESS_KEY=Eby8vdM02xNOcqFEY3PL4kAzqT0qY5cV

**Comment by batuhanbag** [comment]
you need to add these credentials GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET your .env file for authentication and after that app will redirect to github

**Comment by amir-ben-shimol** [comment]
Make sure to add the SERVER_URL configuration under your `.env` file.


_`SERVER_URL`: The URL of your server, for local deployment it will be either http://localhost:3000 or https://localhost:8443. For Azure it will be your Azure App URL_

**Comment by houjinyun** [comment]
> I'm also facing the same issue, any solution?
> 
> Error on page at http://localhost:3000/auth/callback/github?code=someValue&state=someValue
> 
> > Internal Server Error
> 
> Error in logs:
> 
> > Error: connect ECONNREFUSED 127.0.0.1:10002
> 
> My .env has following values:
> 
> EMULATED=true SERVER_URL=http://localhost:3000 GITHUB_CLIENT_ID=someValue GITHUB_CLIENT_SECRET=someValue LOGGING=true

This solve my same problem.

**Comment by icarusdust** [comment]
> I'm able to resolve the **Error: connect ECONNREFUSED 127.0.0.1:10002** issue by installing azurite and running the server locally.
> 
> Just modify **.env** to have below config, which is default for Azurite (emulator for local)
> 
> AZURE_STORAGE_ACCOUNT=devstoreaccount1 AZURE_STORAGE_ACCESS_KEY=Eby8vdM02xNOcqFEY3PL4kAzqT0qY5cV

I do not think that it is necessary to set these values when working with the local storage

**Comment by icarusdust** [comment]
1. `.env` inside ./api directory has the following values:

```
EMULATED=true
SERVER_URL=http://localhost:3000
GITHUB_CLIENT_ID=value
GITHUB_CLIENT_SECRET=value
```

2. Server is up and running at http://localhost:3000
3. Running `code-push-standalone register` opens a browser for authentication, trying to authenticate with Github, it does open a page to authorize, but as soon as clicking to authorize the user, it throws` Internal Server Error`. Upon checking the server-side logs, it is giving `Error: connect ECONNREFUSED 127.0.0.1:10002`

Working with Azurite in VS Code and tried to change the azurite tabel port to 10003, but does not help

**Comment by icarusdust** [comment]
When saying it is working for you, do you guys face any other issues or it works as expected ? Sadly, nothing helps resolve Internal Server Error.

**Comment by ahsanparacha02** [comment]
> > I'm also facing the same issue, any solution?
> > Error on page at http://localhost:3000/auth/callback/github?code=someValue&state=someValue
> > > Internal Server Error
> > 
> > 
> > Error in logs:
> > > Error: connect ECONNREFUSED 127.0.0.1:10002
> > 
> > 
> > My .env has following values:
> > EMULATED=true SERVER_URL=http://localhost:3000 GITHUB_CLIENT_ID=someValue GITHUB_CLIENT_SECRET=someValue LOGGING=true
> 
> This solve my same problem.

I am getting same error with github login as mentioned .env configuration.

![Image](https://github.com/user-attachments/assets/03bcdd42-ada9-4f1b-96be-dbd3c6e039b1)

**Comment by iiifinal** [comment]
Try to clear the data of azurite. Then re-initialize and run azurite. Then change const devConnectionString = "UseDevelopmentStorage=true";
to a full connection string. And do not use the default storage account. And set the value of the storage account in the environment variable configuration file .env or .test.env.
Example: const devConnectionString = "DefaultEndpointsProtocol=http;AccountName=alice;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1: 10000/alice;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;TableEndpoint=http://127.0.0.1:10002/alice;";

**Comment by amiinous** [comment]
I had the same issue and managed to fix it by installing the Azurite via terminal (instead of previously VSCode) by `npm install -g azurite`. Then runnning the command `azurite` (without the any flag like `--silent` or `--location`) to fire up the service.

**Comment by mustafauyysl** [comment]
You must to add Bearer token to your requests. 

Example:

`curl --location 'http://localhost:3000/apps' \
--header 'Authorization: Bearer YOUR BEARER TOKEN'`

You can take your bearer token after register(localhost:3000/auth/register) in chrome.

**Comment by kaushil111** [comment]
I had `DEBUG_DISABLE_AUTH=true` and `DEBUG_USER_ID=1234567890`, removing those allowed me to register.
